Backends/CLX/medium.lisp

```diff
@@ -577,49 +610,47 @@ translated, so they begin at different position than [0,0])."))
                                (medium-sheet medium))
                               x y)
     (with-clx-graphics () medium
-      (cond ((< (line-style-thickness line-style) 2)
-             (let ((x (round-coordinate x))
-                   (y (round-coordinate y)))
-               (when (and (typep x '(signed-byte 16))
-                          (typep y '(signed-byte 16)))
-                 (xlib:draw-point mirror gc x y))))
-            (t
-             (let* ((radius (/ (line-style-thickness line-style) 2))
-                    (min-x (round-coordinate (- x radius)))
-                    (min-y (round-coordinate (- y radius)))
-                    (max-x (round-coordinate (+ x radius)))
-                    (max-y (round-coordinate (+ y radius))))
-               (when (and (typep min-x '(signed-byte 16))
-                          (typep min-y '(signed-byte 16)))
-                   (xlib:draw-arc mirror gc min-x min-y
-                                  (- max-x min-x) (- max-y min-y)
-                                  0 (* 2 pi) t))))))))
+      (let ((radius (/ (line-style-effective-thickness line-style medium) 2)))
```

Comment:
This seems to round+clamp, then divide by 2, then round again, whereas the previous code only rounded in the end. Is this intentional?

Reply:
it is intentional. clamp + round in line-style-effective-thickness (and dashes) is for all lines - clx signalled errors when the value was not a) rounded to integer, b) integer fitting in a byte (protocol thing). This radius is for points, from the center of the point we draw it with such radius (which also needs to be an integer), since value is already clamped there is no need to clamp it further. For instance radius for the effective thickness 33 would be 33/2, which is not an integer.

Reply:
Rounding and clamping only once at the end seems to achieve those things. I'm not sure I understand.

Reply:
me neither, could you show me how do you imagine the code the way you want it to look?

Reply:
For `-draw-point*`, I guess it would be
```lisp
(let ((unrounded-unclamped-circumference (backend-independent-line-style-effective-thickness line-style medium)))
  (if (< unrounded-unclamped-circumference 2)
    ;; draw points without computing radius
    (let ((radius (/ unrounded-unclamped-circumference 2))) ; maybe also round here
      ;; draw arcs using radius
      )))
```

Note that there would be no clamping when drawing points. I assume clamping to `[1,255]` is intended for the line drawing functions, but the point drawing functions use `xlib:draw-arc` which is probably not limited to radii below 255.

The other difference is no rounding and division if the radius is below 1.

I'm not sure whether rounding is needed where it says "maybe also round here". That's a matter of intended behavior. For that matter, the points could be drawn with
```lisp
(xlib:draw-arc … (round-coordinate (- x radius)) (round-coordinate (- x radius)) rounded-circumference rounded-circumference …)
```
but that would result in yet another behavior.